src/portal/lib/src/create-edit-rule/create-edit-rule.component.ts

```diff
@@ -142,6 +142,12 @@ export class CreateEditRuleComponent implements OnInit, OnDestroy {
   }
 
   pullModeChange(): void {
+
+    // add the default value to fix src_registry cannot select when this.sourceList.length==1 in safari;
+    if (this.sourceList.length) {
```

Comment:
this condition is not enough to make sure sourceList length == 1

Reply:
I have change the comments